src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProviderVirtualMachineUtil.java

```diff
@@ -84,9 +72,110 @@ public VirtualMachine searchVirtualMachineByName(String name, Folder rootFolder)
 
     }
 
-    public Folder searchFolderByName(String name, Folder rootFolder)
-            throws InvalidProperty, RuntimeFault, RemoteException {
+    public Folder searchFolderByName(String name, Folder rootFolder) throws RemoteException {
         return (Folder) new InventoryNavigator(rootFolder).searchManagedEntity("Folder", name);
     }
 
+    public HostSystem searchHostByName(String name, Folder rootFolder) throws RemoteException {
+        return (HostSystem) new InventoryNavigator(rootFolder).searchManagedEntity("HostSystem", name);
+    }
+
+    public Folder searchVMFolderFromVMName(String name, Folder rootFolder) throws RemoteException {
+        ManagedEntity current = searchVirtualMachineByName(name, rootFolder).getParent();
+        while (!(current instanceof Folder)) {
```

Comment:
You walk through a tree in which you need to downcast each element. It doesn't seem right. I went through the code from the MangedEntity and didn't find another way. I was hoping to be able to access the type attribute somewhere, without instanceof. Maybe the tags? 

You could use the searchManagedEntities(String[][] typeinfo, boolean recurse)

Maybe you have another idea?